packages/grid/_modules_/grid/hooks/utils/useOptionsProp.ts

```diff
@@ -24,6 +25,7 @@ export function useOptionsProp(apiRef: ApiRef, props: GridComponentProps): GridO
     () => ({
       ...props,
       localeText: { ...DEFAULT_LOCALE_TEXT, ...props.localeText },
+      scrollbarSize: props.scrollbarSize == null ? getScrollbarSize(document) : props.scrollbarSize,
```

Comment:
I don't think that we can run the logic at each render. It would expect it to seriously degrade the performance (looking at what getScrollbarSize does). I don't have a benchmark to prove it. What about a memo per component mount lifecycle?

Reply:
It also can't work with server-side rendering